iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/plan/planner/LocalExecutionPlanContext.java

```diff
@@ -91,6 +98,10 @@ public class LocalExecutionPlanContext {
   // use AtomicReference not for thread-safe, just for updating same field in different pipeline
   private AtomicReference<List<Long>> timePartitions = new AtomicReference<>();
 
+  /** Records the parent of each pipeline. The order of each list does not matter for now. */
+  private Map<PlanNodeId, List<PipelineMemoryEstimator>> parentPlanNodeIdToMemoryEstimator =
```

Comment:
remember to set this map to null, after you check the memory, this Map may occupy huge memory when there are many devices;

Reply:
Done.